src/contracts/gasCalculator/BaseGasCalculator.sol

```diff
@@ -0,0 +1,63 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.25;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IL2GasCalculator } from "src/contracts/interfaces/IL2GasCalculator.sol";
+
+/// @notice Implementation:
+/// https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/GasPriceOracle.sol
+/// @notice Deployment on Base: https://basescan.org/address/0x420000000000000000000000000000000000000f
+interface IGasPriceOracle {
+    function getL1FeeUpperBound(uint256 _unsignedTxSize) external view returns (uint256);
+}
+
+contract BaseGasCalculator is IL2GasCalculator, Ownable {
+    uint256 internal constant _CALLDATA_LENGTH_PREMIUM = 32;
+    uint256 internal constant _BASE_TRANSACTION_GAS_USED = 21_000;
+
+    address public immutable gasPriceOracle;
```

Comment:
Think immutable should be capitalised like constants to show it can't be changed once set.
```suggestion
    address public immutable GAS_PRICE_ORACLE;
```